src/config_directives.c

```diff
@@ -164,16 +164,36 @@ CFGFUN(for_window, const char *command) {
     TAILQ_INSERT_TAIL(&assignments, assignment, assignments);
 }
 
-void create_gaps_assignment(const char *workspace, bool inner, const long value) {
+void create_gaps_assignment(const char *workspace, const char *scope, gaps_t gaps) {
     DLOG("Setting gaps for workspace %s", workspace);
 
     struct Workspace_Assignment *assignment;
     TAILQ_FOREACH(assignment, &ws_assignments, ws_assignments) {
         if (strcasecmp(assignment->name, workspace) == 0) {
-            if (inner)
-                assignment->gaps.inner = value;
-            else
-                assignment->gaps.outer = value;
+            if (!strcmp(scope, "inner"))
+                assignment->gaps.inner = gaps.inner;
+            else if (!strcmp(scope, "outer")) {
+                assignment->gaps.top = gaps.top;
+                assignment->gaps.right = gaps.right;
+                assignment->gaps.bottom = gaps.bottom;
+                assignment->gaps.left = gaps.left;
+            } else if (!strcmp(scope, "vertical")) {
+                assignment->gaps.top = gaps.top;
+                assignment->gaps.bottom = gaps.bottom;
+            } else if (!strcmp(scope, "horizontal")) {
+                assignment->gaps.right = gaps.right;
+                assignment->gaps.left = gaps.left;
+            } else if (!strcmp(scope, "top"))
+                assignment->gaps.top = gaps.top;
+            else if (!strcmp(scope, "right"))
```

Comment:
Here and following, please also use `{}`. Also further down in the changes of `create_gaps_assignment`.